src/http/common.cr

```diff
@@ -95,9 +109,28 @@ module HTTP
       true
     end
   end
+
+  def self.parse_time(time_str : String)
+    patterns = ["%a, %d %b %Y %H:%M:%S %z", "%A, %d-%b-%y %H:%M:%S %z", "%a %b %e %H:%M:%S %Y"]
+    patterns.each do |pattern|
+      begin
+        return Time.parse(time_str, pattern)
+      rescue
+        # TODO: do something here?
+      end
+    end
+
+    return nil
+  end
+
+  def self.rfc1123_string(time : Time) : String
```

Comment:
Isn't RFC1123 stating that dates must be formatted as per RFC822? How about having generic methods like `Time#to_rfc822` and `Time.from_rfc822(str)` instead?